### PR TITLE
[POC-WIP] Sending mails use-case can save InputStream instead of Bytes

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
[09485fb216fd577d2e780296a6729ca844255a05] [INFO] Running org.apache.james.blob.mail.MimeMessageStoreTest
[09485fb216fd577d2e780296a6729ca844255a05] 13:13:37.887 [ERROR] r.c.s.Schedulers - Scheduler worker in group main failed with an uncaught exception
```
as expected

File: server/blob/mail-store/src/main/java/org/apache/james/blob/mail/MimeMessageStore.java [hunk 1]
```diff
@@ -70,59 +73,29 @@ public Factory(BlobStore blobStore) {
...
-        }
 
-        private static byte[] getHeaderBytes(byte[] messageContentAsArray, int bodyStartOctet) {
-            ByteBuffer headerContent = ByteBuffer.wrap(messageContentAsArray, 0, bodyStartOctet);
-            byte[] headerBytes = new byte[bodyStartOctet];
-            headerContent.get(headerBytes);
-            return headerBytes;
-        }
+        private static MessageParts readMessage(MimeMessage message) throws IOException {
+            PipedOutputStream outputStream = new PipedOutputStream();
```
Comment (chibenwa) [inline server/blob/mail-store/src/main/java/org/apache/james/blob/mail/MimeMessageStore.java]: I'm not sure we need "piping" here (which is blocking one thread here) while we could write an outputstream version directly...

File: server/container/util/src/main/java/org/apache/james/util/BodyOffsetInputStream.java [hunk 2]
```diff
@@ -30,8 +37,64 @@
...
+                                                  PipedInputStream headerInputStream,
+                                                  PipedInputStream bodyInputStream) throws IOException {
+            try (BodyOffsetInputStream offsetInputStream = inputStream;
+                 PipedOutputStream headerOutputStream = new PipedOutputStream(headerInputStream);
+                 PipedOutputStream bodyOutputStream = new PipedOutputStream(bodyInputStream)) {
+
+                int currentByte;
+                while ((currentByte = offsetInputStream.read()) != END_OF_STREAM) {
+                    if (offsetInputStream.bodyStartOctet == NOT_FOUND
+                        || offsetInputStream.readBytes <= offsetInputStream.bodyStartOctet) {
```
Comment (chibenwa) [inline server/container/util/src/main/java/org/apache/james/util/BodyOffsetInputStream.java]: Name your boolean conditions ;-)

File: server/container/util/src/main/java/org/apache/james/util/BodyOffsetInputStream.java [hunk 3]
```diff
@@ -30,8 +37,64 @@
...
+            }
+        }
+
+        public static MessageParts split(BodyOffsetInputStream inputStream) {
+            PipedInputStream headerInputStream = new PipedInputStream();
+            PipedInputStream bodyInputStream = new PipedInputStream();
+
+            Mono.fromRunnable(Throwing.runnable(() -> consumeOriginalStream(inputStream, headerInputStream, bodyInputStream)))
+                .subscribeOn(Schedulers.elastic())
+                .subscribe();
```
Comment (chibenwa) [inline server/container/util/src/main/java/org/apache/james/util/BodyOffsetInputStream.java]: Too much piping-sception to me...

**Comment by Arsnael** [comment]
Read it

**Comment by trantienduchn** [comment]
It doesn't work

**Pull request closed without merging**